libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -0,0 +1,1000 @@
+
+   key_type ret;
+   ret.resize(buffer_size);
+
+   memcpy(ret.data(), &status, sizeof(status));
+   memcpy(ret.data() + sizeof(status), &bige_table, index_name_size);
+   memcpy(ret.data() + sizeof(status) + index_name_size, &bige_index, index_name_size);
+
+   return ret;
+}
+
+inline key_type table_key(const key_type& prefix, const key_type& key) {
+   return prefix + key;
+}
+
+template <typename I>
+inline I flip_msb(I val) {
+   constexpr static size_t bits = sizeof(I) * 8;
+   return val ^ (static_cast<I>(1) << (bits - 1));
+}
+
+template <typename I>
+inline I get_msb(I val) {
+   constexpr static size_t bits = sizeof(I) * 8;
+   return val >> (bits - 1);
+}
+
+template <typename I, typename std::enable_if_t<std::is_integral<I>::value, int> = 0>
+inline key_type make_key(I val) {
+   if (std::is_signed<I>::value) {
+      val = flip_msb(val);
+   }
+
+   auto big_endian = swap_endian<I>(val);
+
+   const char* bytes = reinterpret_cast<char*>(&big_endian);
+   constexpr size_t size = sizeof(big_endian);
+   key_type s(bytes, size);
+   return s;
+}
+
+template <typename I, typename F>
+inline key_type make_floating_key(F val) {
+   if (val == -0) {
+      val = +0;
+   }
+
+   auto* ival = reinterpret_cast<I*>(&val);
+   I bit_val;
+   auto msb = get_msb(*ival);
+   if (msb) {
+      // invert all the bits
+      bit_val = ~(*ival);
+   } else {
+      // invert just msb
+      bit_val = flip_msb(*ival);
+   }
+
+   auto big_endian = swap_endian<I>(bit_val);
+
+   const char* bytes = reinterpret_cast<char*>(&big_endian);
+   constexpr size_t size = sizeof(big_endian);
+   key_type s(bytes, size);
+   return s;
+}
+
+inline key_type make_key(float val) {
+   return make_floating_key<uint32_t>(val);
+}
+
+inline key_type make_key(double val) {
+   return make_floating_key<uint64_t>(val);
+}
+
+inline key_type make_key(const char* str, size_t size, bool case_insensitive=false) {
+   size_t data_size = size + 3;
+   void* data_buffer = data_size > detail::max_stack_buffer_size ? malloc(data_size) : alloca(data_size);
+
+   if (case_insensitive) {
+      std::transform(str, str + size, (char*)data_buffer, [](unsigned char c) -> unsigned char { return std::toupper(c); });
+   } else {
+      memcpy(data_buffer, str, size);
+   }
+
+   ((char*)data_buffer)[data_size - 3] = 0x01;
+   ((char*)data_buffer)[data_size - 2] = 0x00;
+   ((char*)data_buffer)[data_size - 1] = 0x00;
+
+   key_type s((const char*)data_buffer, data_size);
+
+   if (data_size > detail::max_stack_buffer_size) {
+      free(data_buffer);
+   }
+   return s;
+}
+
+inline key_type make_key(const std::string& val, bool case_insensitive=false) {
+   return make_key(val.data(), val.size(), case_insensitive);
+}
+
+inline key_type make_key(const std::string_view& val, bool case_insensitive=false) {
+   return make_key(val.data(), val.size(), case_insensitive);
+}
+
+inline key_type make_key(const char* str, bool case_insensitive=false) {
```

Comment:
I didn't bother reviewing any of the `make_key` stuff (or any of the functions that only exist to implement `make_key` functions such as `swap_endian`, `flip_msb`, ` get_msb`, and `make_floating_key`) because from what I understood all this will be replaced by abieos. 

But I also want to confirm that we are going to be removing all the case insensitive stuff, right?

Reply:
Yes, all case insensitive stuff will be removed.